lib/ansible/modules/packaging/os/rhsm_repository.py

```diff
@@ -0,0 +1,260 @@
+'''
+
+RETURN = '''
+repositories:
+  description:
+    - The list of RHSM repositories with their states.
+    - When this module is used to change the repositories states, this list contains the updated states after the changes.
+  returned: success
+  type: list
+'''
+
+import re
+import os
+from fnmatch import fnmatch
+from copy import deepcopy
+from ansible.module_utils.basic import AnsibleModule
+
+
+def run_subscription_manager(module, arguments):
+    # Execute subuscription-manager with arguments and manage common errors
+    rhsm_bin = module.get_bin_path('subscription-manager')
+    if not rhsm_bin:
+        module.fail_json(msg='The executable file subscription-manager was not found in PATH')
+
+    rc, out, err = module.run_command("%s %s" % (rhsm_bin, " ".join(arguments)))
+
+    if rc == 1 and (err == 'The password you typed is invalid.\nPlease try again.\n' or os.getuid() != 0):
+        module.fail_json(msg='The executable file subscription-manager must be run using root privileges')
+    elif rc == 0 and out == 'This system has no repositories available through subscriptions.\n':
+        module.fail_json(msg='This system has no repositories available through subscriptions')
+    elif rc == 1:
+        module.fail_json(msg='subscription-manager failed with the following error: %s' % err)
+    else:
+        return rc, out, err
+
+
+def get_repository_list(module, list_parameter):
+    # Generate RHSM repository list and return a list of dict
+    if list_parameter == 'list_enabled':
+        rhsm_arguments = ['repos', '--list-enabled']
+    elif list_parameter == 'list_disabled':
+        rhsm_arguments = ['repos', '--list-disabled']
+    elif list_parameter == 'list':
+        rhsm_arguments = ['repos', '--list']
+    rc, out, err = run_subscription_manager(module, rhsm_arguments)
+
+    skip_lines = ['+----------------------------------------------------------+',
+                  '    Available Repositories in /etc/yum.repos.d/redhat.repo']
+    repo_id_re_str = r'Repo ID:   (.*)'
+    repo_name_re_str = r'Repo Name: (.*)'
+    repo_url_re_str = r'Repo URL:  (.*)'
+    repo_enabled_re_str = r'Enabled:   (.*)'
+
+    repo_id = ''
+    repo_name = ''
+    repo_url = ''
+    repo_enabled = ''
+
+    repo_result = []
+
+    for line in out.split('\n'):
+        if line in skip_lines:
+            continue
+
+        repo_id_re = re.match(repo_id_re_str, line)
+        if repo_id_re:
+            repo_id = repo_id_re.group(1)
+            continue
+
+        repo_name_re = re.match(repo_name_re_str, line)
+        if repo_name_re:
+            repo_name = repo_name_re.group(1)
+            continue
+
+        repo_url_re = re.match(repo_url_re_str, line)
+        if repo_url_re:
+            repo_url = repo_url_re.group(1)
+            continue
+
+        repo_enabled_re = re.match(repo_enabled_re_str, line)
+        if repo_enabled_re:
+            repo_enabled = repo_enabled_re.group(1)
+
+            repo = {"id": repo_id,
```

Comment:
Please format this like this:

```python
            repo = {
                "id": repo_id,
                "name": repo_name,
                "url": repo_url,
                "enabled": True if repo_enabled == '1' else False
            }
```

Reply:
Changed in the last commit, thanks for the review